apps/src/applab/PlaySpaceHeader.jsx

```diff
@@ -0,0 +1,106 @@
+/** @file Row of controls above the visualization. */
+// Strict linting: Absorb into global config when possible
+/* jshint
+ unused: true,
+ eqeqeq: true,
+ maxlen: 120
+ */
+
+var constants = require('./constants');
+var msg = require('../locale');
+var ScreenSelector = require('./ScreenSelector.jsx');
+var ToggleButton = require('./ToggleButton.jsx');
+var ToggleGroup = require('./ToggleGroup.jsx');
+var ViewDataButton = require('./ViewDataButton.jsx');
+
+var Mode = constants.MODE;
+
+var PlaySpaceHeader = React.createClass({
+  propTypes: {
+    hideToggle: React.PropTypes.bool.isRequired,
+    hideViewDataButton: React.PropTypes.bool.isRequired,
+    startInDesignMode: React.PropTypes.bool.isRequired,
+    initialScreen: React.PropTypes.string.isRequired,
+    screenIds: React.PropTypes.array.isRequired,
+    onDesignModeButton: React.PropTypes.func.isRequired,
+    onCodeModeButton: React.PropTypes.func.isRequired,
+    onViewDataButton: React.PropTypes.func.isRequired,
+    onScreenChange: React.PropTypes.func.isRequired,
+    onScreenCreate: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      mode: this.props.startInDesignMode ? Mode.DESIGN :  Mode.CODE,
+      activeScreen: null
+    };
+  },
+
+  handleSetMode: function (newMode) {
+    if (this.state.mode === newMode) {
+      return;
+    }
+    if (newMode === Mode.CODE) {
+      this.props.onCodeModeButton();
+    } else {
+      this.props.onDesignModeButton();
+    }
+
+    this.setState({
+      mode: newMode
+    });
+  },
+
+  handleScreenChange: function (evt) {
+    var screenId = evt.target.value;
+    if (screenId === constants.NEW_SCREEN) {
+      screenId = this.props.onScreenCreate();
+    }
+    this.props.onScreenChange(screenId);
+  },
+
+  componentWillReceiveProps: function (newProps) {
+    this.setState({ activeScreen: newProps.initialScreen });
+  },
+
+  render: function () {
+    var leftSide, rightSide;
+
+    if (!this.props.hideToggle) {
+      leftSide = (
+          <ToggleGroup selected={this.state.mode} onChange={this.handleSetMode}>
+            <ToggleButton
+                id='codeModeButton'
+                value={Mode.CODE}>{msg.codeMode()}</ToggleButton>
+            <ToggleButton
+                id='designModeButton'
+                value={Mode.DESIGN}>{msg.designMode()}</ToggleButton>
+          </ToggleGroup>
```

Comment:
Does that potentially introduce whitespace inside the element, or does JSX magically clean that up for us?  I know in the past, at least writing straight-up HTML, I have to avoid newlines between an opening tag and its content if I'm being precise about the output.